docs/tutorials/developer-journey/level-2/2.5-unit-testing.md

```diff
@@ -0,0 +1,406 @@
+
+- In the suite, 2 tests are defined. One tests if the `Nat` value `10` is equal to `10`. The second tests if the `Nat` value of `5` is greater than `3`. 
+
+- Lastly, the suite is run with the line `Suite.run(suite)`.
+
+### Canister unit testing
+In addition to the `Suite`, `Testable` and `Matchers` packages, Motoko Matchers also contains a `Canister` package. The `Canister` package can be used to execute unit tests for canisters. Below is an example that uses this `Canister` package:
+
+```motoko
+import Canister "canister:CanisterName";
+import C "mo:matchers/Canister";
+import M "mo:matchers/Matchers";
+import T "mo:matchers/Testable";
+
+actor {
+    let it = C.Tester({ batchSize = 8 });
+    public shared func test() : async Text {
+
+        it.should("greet me", func () : async C.TestResult = async {
+          let greeting = await Canister.greet("Bob");
+          M.attempt(greeting, M.equals(T.text("Hello, Bob!")))
+        });
+
+        it.shouldFailTo("greet my friend", func () : async () = async {
+          let greeting = await Canister.greet("George");
+          ignore greeting
+        });
+
+        await it.runAll()
+        // await it.run()
+    }
+}
+```
+
+This unit test does the following:
+
+- First, it imports a canister called `CanisterName`. This allows the result of functions within the canister to be tested. 
+
+- Then, it imports the `Canister`, `Testable` and `Matchers` packages from the Motoko Matchers project. 
+
+- A `batchSize` is defined; this determines how many times the test will be run. 
+
+- Two tests are defined. Both test the result of the canister's `greet` function: 
+
+    - The first tests if the canister's greeting function is passed the text 'Bob', does the result text equal 'Hello, Bob!'?
+
+    - The second tests if the greeting doesn't greet 'Bob' and instead greets 'George', the greeting should be ignored. 
+
+- Then, it runs all tests in the batch, indicated by the `runAll()` call. 
+
+Want to look at a more complex, running example? You can take a look at the unit tests that are used in the [invoice canister example](https://github.com/dfinity/invoice-canister/tree/main/test).
+
+Want to learn more about running large, long-running batches of tests? Take a look at the [Motoko BigTest library.](https://github.com/matthewhammer/motoko-bigtest)
+
+## Rust unit testing
+
+In this developer journey series, we'll continue to focus on exploring and using Motoko. However, it's important to make note of the resources available for Rust testing.
+
+For information about Rust unit testing, see the documentation [here](https://mmapped.blog/posts/01-effective-rust-canisters.html#test-upgrades) and [here](https://mmapped.blog/posts/01-effective-rust-canisters.html#target-independent). 
+
+You may also consider the [DFINITY Rust guidelines on testing](https://docs.dfinity.systems/dfinity/spec/meta/rust.html#_tests).
+
+### Tests using Light Replica
```

Comment:
@michael-weigelt It might make sense to mention/reference PocketIC here as well.

Reply:
Yes. Should we write a paragraph, or how does this work? 

@dfinity/testing-verification should we wait until we have released a rust library?

Reply:
Short answer: I'd say yes.

Long answer: You wouldn't want people who principally use Rust to choose something else now only to discover later that Rust is actually an option.

Reply:
I've added a paragraph describing PocketIC and linking to the PocketIC repo.